src/Cryptor.ts

```diff
@@ -1,131 +1,160 @@
-import * as fs from 'fs';
-import * as crypto from 'crypto';
-import * as process from 'process';
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import * as process from 'process'
+import { spawn, Worker, ModuleThread } from 'threads'
+import { CryptorWorker } from './CryptorWorker'
 
 // TODO: function docs
 
+// interface CryptorParameters {
+// 	key: Buffer | string
+// 	initVector?: Buffer
+// 	algorithm?: string
+// 	useWebWorkers?: boolean
+// }
+
 export default class Cryptor {
-  private _algo: string;
-  private readonly _initVector: Buffer;
-  private _key: Buffer;
-  private _cipher: crypto.Cipher;
-  private _decipher: crypto.Decipher;
-	constructor(pass: string, initVector: Buffer = null, algo: string = 'id-aes256-GCM') {
-		this._algo = algo;
-    	this._initVector = initVector ? initVector : this.genRandomIVSync();
+	private _algorithm: string
+	private _initVector: Buffer
+	private _key: Buffer
+	private _cipher: crypto.Cipher
+	private _decipher: crypto.Decipher
+	private _useWebWorkers: boolean
```

Comment:
If you use `private` scoping. Do you need to use `_` as a prefix? Or is that convention in typescript? If it's not convention in TS, then private scope is enough without needing to also use `_` as a prefix.

Reply:
Microsoft recommends against using underscores: https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines
so I will remove them